lib/doctest.mjs

```diff
@@ -0,0 +1,62 @@
+import {readFile, writeFile, unlink} from 'fs';
+import {resolve} from 'path';
+import {promisify} from 'util';
+
+import common from './common';
+import doctest from './doctest.js';
+
+export default async function(path, options) {
+  if (options.module === 'esm') {
+    if (options.type != null) {
+      throw new Error ('Cannot use file type when module is "esm"');
+    }
+  } else {
+    return doctest (path, options);
+  }
```

Comment:
What do you think of this approach, which limits nesting?

```javascript
if (options.module !== 'esm') return doctest (path, options);

if (options.type != null) {
  throw new Error ('Cannot use file type when module is "esm"');
}
```

It's just an idea. Stick with what you have if you prefer it. :)

Reply:
I like it :zap: